File: 4-median-of-two-sorted-arrays/4-median-of-two-sorted-arrays.java
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        Arrays.sort(nums1);
        Arrays.sort(nums2);
        
        int i=0 , j=0, count;
        double m1=-1, m2=-1;
        int n= nums1.length;
        int m= nums2.length;
        if((m+n)%2==1){
            for(count=0; count<=(m+n)/2; count++){
                if(i != n && j !=m){
                    m1= (nums1[i] >nums2[j]) ? nums2[j++] : nums1[i++];
                }
                else if(i<n){
                    m1= nums1[i++];
                }
                else{
                    m1=nums2[j++];
                }
            }
            return m1;
        }
        
        else{
            for(count=0; count<= (m+n)/2; count++){
                m2=m1;
                if(i !=n && j!=m){
                    m1=(nums1[i] > nums2[j]) ? nums2[j++] : nums1[i++];
                }
                else if(i<n){
                    m1=nums1[i++];
                }
                else{
                    m1=nums2[j++];
                }
            }
            return (m1 + m2)/2;
        }
        
    }
}

File: 4-median-of-two-sorted-arrays/4-median-of-two-sorted-arrays.cpp
class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
         for(auto n:nums2)
        {
            nums1.push_back(n);
            
        }
        int size = nums1.size();
        sort(nums1.begin(),nums1.end());
        
        /*int mid=nums1.size()/2;
        if(nums1.size()%2==0)
        {
            return (double)(mid+(mid-1))/2;
        }
        else
        {
            return (double)(mid-1);
        }*/
        
         if(size%2==1){
             return nums1[size/2];
        }
        else{
            return ((double)nums1[size/2 -1] + (double)nums1[size/2])/2;
        }
    }
};